nomad/structs/structs.go

```diff
@@ -943,7 +943,10 @@ type JobRevertRequest struct {
 	JobID string
 
 	// JobVersion the version to revert to.
-	JobVersion uint64
+	JobVersion *uint64
```

Comment:
You can't change the type of fields in RPC structs, because they can be in-flight across upgrades (which can take hours or even days for weirdly cautious users). We should be fixing this in the CLI/UI like we did for untag.

Reply:
Hrm, you _might_ be able to get away with this if the encoding works across upgrades actually... but you'd need to test it with production binaries for both older and newer to be sure. It's really not worth the risk, and having pointer fields on these in the RPC handlers is a bit of a red-flag in terms of design anyways.

Reply:
Personally I'm not a fan of pointers to basic types in general.

What if neither were pointers and the contract was ~

```go
if JobVersionTag != "" {
	if JobVersion != 0 {
		return errors.New("can't have it both ways, bud")
	}
	// use JobVersionTag
} else {
	// use JobVersion
}
```

but I suppose if the CLI just does a lookup, then everything else on the backend can just deal in version uints as usual, so this PR could be limited to `command/agent/job_revert*.go`

Reply:
Yeah, which is pretty much what we did for untag.